prov/usnic/src/usdf.h

```diff
@@ -326,16 +326,31 @@ struct usdf_mr {
 	struct usd_mr *mr_mr;
 };
 
+struct usdf_soft_comp {
+	int status;
+	/* Add new fields as required */
+};
+
 struct usdf_cq_hard {
 	struct usdf_cq *cqh_cq;
 	struct usd_cq *cqh_ucq;
 	atomic_t cqh_refcnt;
 	void (*cqh_progress)(struct usdf_cq_hard *hcq);
-	void (*cqh_post)(struct usdf_cq_hard *hcq, void *context, size_t len);
+	void (*cqh_post)(struct usdf_cq_hard *hcq, void *context, size_t len,
+			struct usdf_soft_comp soft_comp);
```

Comment:
Let's not pass a struct as a parameter by value; things will get ugly when we expand the size of this struct.  Let's just add a new `prov_errno` argument instead that should be an `FI_` error code.  When we add other information in the future (e.g., `buf`), we can add more arguments to `cqh_post` at that time.